algorithms.py

```diff
@@ -825,6 +890,160 @@ def hudson_recombination_event(self, label, return_heads=False):
             ret = x, z
         return ret
 
+    def wiuf_geneconversion_within_event(self, label, return_heads=False):
+        """
+        Implements a gene conversion event that starts within a segment
+        """
+        h = random.randint(1, self.L[label].get_total())
+        # generate tracklength
+        tl = np.random.geometric(1/self.tracklength)
+        # Get the segment containing the h'th link
+        y = self.segments[self.L[label].find(h)]
+        k = y.right - self.L[label].get_cumulative_frequency(y.index) + h - 1
+        # check if the gene conversion falls between segments --> no effect
+        if y.left >= k+tl:
+            # print("noneffective GCI EVENT")
+            return None
+        self.num_gc_events += 1
+        x = y.prev
+        # both breaks are within the same segment
+        if k+tl < y.right:
+            if k <= y.left:
+                y.prev = None
+                z2 = self.alloc_segment(
+                    k+tl, y.right, y.node, y.population, x, y.next)
+                if x is not None:
+                    x.next = z2
+                    lhs = x.right
+                    lhs_tail = x
+                else:
+                    print("this should never be possible")
```

Comment:
An assert would be better here. Also, why the line of code immediately after?

Reply:
I changed this to an assert.
This case is not impossible but handled in a different branch of the algorithm. The line of code immediately after would be necessary if the case would get to this point.